src/org/opensolaris/opengrok/web/suggester/parser/SuggesterQueryDataParser.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved.
+ */
+package org.opensolaris.opengrok.web.suggester.parser;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.search.Query;
+import org.opengrok.suggest.query.SuggesterQuery;
+import org.opensolaris.opengrok.search.QueryBuilder;
+import org.opensolaris.opengrok.web.suggester.model.SuggesterData;
+import org.opensolaris.opengrok.web.suggester.model.SuggesterQueryData;
+import org.opensolaris.opengrok.web.suggester.query.SuggesterQueryBuilder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Parser for the raw {@link SuggesterQueryData}.
+ */
+public class SuggesterQueryDataParser {
+
+    private static final int IDENTIFIER_LENGTH = 5;
+
+    private static final Logger logger = Logger.getLogger(SuggesterQueryDataParser.class.getName());
```

Comment:
No `.class` in `getLogger()` -> https://docs.oracle.com/javase/8/docs/api/java/util/logging/Logger.html

I cannot use OpenGrok's `LoggerFactory.getLogger` since this module has no dependency to it.

Reply:
Sorry, it's not :D (I somehow got confused) I will repair it :)

Reply:
:D